ddtrace/propagation/http.py

```diff
@@ -120,24 +120,24 @@ def _inject(span_context, headers):
         if span_context.dd_origin is not None:
             headers[HTTP_HEADER_ORIGIN] = ensure_text(span_context.dd_origin)
 
+        # Skip injecting tracer tags
+        if not config._propagation_datadog_tags_enabled:
+            return
+
         # Do not try to encode tags if we have already tried and received an error
         if "_dd.propagation_error" in span_context._meta:
             return
 
-        # TODO: add whitelist
-        # Only propagate tags that start with `_dd.p.`
-        tags_to_encode = {}  # type: Dict[str, str]
-        for key, value in span_context._meta.items():
-            # DEV: encoding will fail if the key or value are not `str`
-            key = ensure_str(key)
-            if key.startswith("_dd.p."):
-                tags_to_encode[key] = ensure_str(value)
+        # Only propagate tags that are allowed for propagation
+        tags_to_encode = {  # type: Dict[str, str]
+            k: v
+            for k, v in ((ensure_str(k), v) for k, v in span_context._meta.items())
+            if k in _DatadogMultiHeader._PROPAGATED_TAGS
```

Comment:
I wonder if we should reverse this:

```python
tags_to_encode = {
    k: span_context._meta[k]
    for k in _DatadogMultiHeader._PROPAGATED_TAGS
    if k in span_context._meta
}
```

Should consistently be less iterations?

Reply:
(micro optimization?)

Reply:
we need to do the `ensure_str` before doing the lookup

Reply:
I would `ensure_str` for the value only if the test on `k` is `True`. WDYT?